src/provider/systemdmanager/service.rs

```diff
@@ -0,0 +1,321 @@
+
+        for (name, value) in env_vars {
+            unit.add_env_var(&name, &value);
+        }
+
+        unit.add_property("Service", "StandardOutput", "journal");
+        unit.add_property("Service", "StandardError", "journal");
+        unit.add_property("Install", "WantedBy", "multi-user.target");
+
+        Ok(unit)
+    }
+
+    pub fn new_from_pod(pod: &Pod) -> Self {
+        let mut unit = SystemDUnit {
+            name: pod.name().to_string(),
+            sections: Default::default(),
+            environment: Default::default(),
+        };
+
+        let restart_policy = match &pod.as_kube_pod().spec {
+            Some(spec) => spec.restart_policy.as_deref().unwrap_or("Never"),
+            None => "Never",
+        };
+
+        unit.add_property(
+            "Service",
+            "Restart",
+            RESTART_POLICY_MAP.get(restart_policy).unwrap(),
+        );
+        unit
+    }
+
+    // Add a key=value entry to the specified section
+    fn add_property(&mut self, section: &'static str, key: &str, value: &str) {
+        let section = self
+            .sections
+            .entry(String::from(section))
+            .or_insert_with(HashMap::new);
+        section.insert(String::from(key), String::from(value));
+    }
+
+    fn add_env_var(&mut self, name: &str, value: &str) {
+        self.environment
+            .insert(String::from(name), String::from(value));
+    }
+
+    pub fn get_unit_file_content(&self) -> String {
+        let mut unit_file_content = String::new();
+
+        // Iterate over all sections and write out its header and content
+        for (section, entries) in &self.sections {
+            unit_file_content.push_str(&format!("[{}]\n", section));
+            for (key, value) in entries {
+                unit_file_content.push_str(&format!("{}={}\n", key, value));
+            }
+            if section.eq("Service") {
+                // Add environment variables to Service section
+                for (name, value) in &self.environment {
+                    unit_file_content.push_str(&format!("Environment=\"{}={}\"\n", name, value));
+                }
+            }
+            unit_file_content.push('\n');
+        }
+        unit_file_content
+    }
+
+    fn get_environment(
+        container: &Container,
+        pod_state: &PodState,
+    ) -> Result<Vec<(String, String)>, StackableError> {
+        // Create template data to be used when rendering template strings
+        let template_data = if let Ok(data) = CreatingConfig::create_render_data(&pod_state) {
+            data
+        } else {
+            error!("Unable to parse directories for command template as UTF8");
+            return Err(PodValidationError {
+                msg: format!(
+                    "Unable to parse directories for command template as UTF8 for container [{}].",
+                    container.name()
+                ),
+            });
+        };
```

Comment:
This is 1:1 copy from get_command. Can this be moved to a method?

Reply:
Potentially :)
I can't think straight any more at the moment, will take another stab later tonight.